02-peer-protocol.md

```diff
@@ -1185,11 +1188,18 @@ The sending node:
   - MUST set `funding_feerate_perkw` to the feerate for this transaction
   - If it requires the receiving node to only use confirmed inputs:
     - MUST set `require_confirmed_inputs`
+  - If it wants the receiving node to contribute to the funding transaction
+    using `option_will_fund`:
+    - MUST send `request_funding` containing one of the `lease_type`s
+      advertised by the receiving node in their [liquidity ads](07-routing-gossip.md#liquidity-ads).
```

Comment:
Thinking about the wording here. Does this imply the `lease_type` used MUST be advertised in gossip? Or can it also be supplied out-of-band?

Reply:
We should allow `lease_type`s to be obtained out-of-band, that should be rephrased to something like:

```
- MUST send `request_funding` containing one of the `lease_type`s supported by the receiving node
```

Reply:
> We should allow `lease_type`s to be obtained out-of-band, that should be rephrased to something like:
> 
> ```
> - MUST send `request_funding` containing one of the `lease_type`s supported by the receiving node
> ```

Sounds good

Reply:
Done in https://github.com/lightning/bolts/pull/1153/commits/357a8b01ae4d63023d8e65f644ead966e367800c